### PR TITLE
Set release version to java 8 to prevent using APIs from newer java versions

### LINKED ISSUE #1066: java.lang.NoSuchMethodError: java.nio.ByteBuffer.rewind()Ljava/nio/ByteBuffer on Atomix::start
**Expected behavior**

The Atomix instance should start correctly

**Actual behavior**

Fail with:
```java
java.lang.NoSuchMethodError: java.nio.ByteBuffer.rewind()Ljava/nio/ByteBuffer;
        at io.atomix.storage.journal.JournalSegmentDescriptor$Builder.build(JournalSegmentDescriptor.java:285)
        at io.atomix.storage.journal.SegmentedJournal.open(SegmentedJournal.java:250)
        at io.atomix.storage.journal.SegmentedJournal.<init>(SegmentedJournal.java:97)
        at io.atomix.storage.journal.SegmentedJournal$Builder.build(SegmentedJournal.java:873)
        at io.atomix.protocols.raft.storage.log.RaftLog$Builder.build(RaftLog.java:309)
        at io.atomix.protocols.raft.storage.RaftStorage.openLog(RaftStorage.java:336)
        at io.atomix.protocols.raft.impl.RaftContext.<init>(RaftContext.java:163)
        at io.atomix.protocols.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:267)
        at io.atomix.protocols.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:228)
        at io.atomix.protocols.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:180)
        at io.atomix.protocols.raft.partition.impl.RaftPartitionServer.start(RaftPartitionServer.java:92)
        at io.atomix.protocols.raft.partition.RaftPartition.open(RaftPartition.java:136)
        at io.atomix.protocols.raft.partition.RaftPartitionGroup.lambda$join$6(RaftPartitionGroup.java:241)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1556)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
        at io.atomix.protocols.raft.partition.RaftPartitionGroup.join(RaftPartitionGroup.java:243)
        at io.atomix.primitive.partition.impl.DefaultPartitionService.lambda$start$1(DefaultPartitionService.java:158)
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
        at io.atomix.primitive.partition.impl.DefaultPartitionService.lambda$start$2(DefaultPartitionService.java:150)
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
        at io.atomix.primitive.partition.impl.DefaultPartitionService.start(DefaultPartitionService.java:138)
        at io.atomix.core.Atomix.lambda$startServices$3(Atomix.java:853)
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966)
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940)
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
        at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```
Due to https://github.com/eclipse/jetty.project/issues/3244#issue-396988904

```

**Environment**

 - Atomix: 3.1.7
 - OS: Linux matrix 4.18.16-300.fc29.x86_64 #1 SMP Sat Oct 20 23:24:08 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
 - JVM OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)

### PR DESCRIPTION
**Pull request by jonohart**
This solves the issue of encountering the exception
`java.lang.NoSuchMethodError: java.nio.ByteBuffer.rewind()Ljava/nio/ByteBuffer` when running on java 8, as per 
https://github.com/eclipse/jetty.project/issues/3244
https://stackoverflow.com/questions/43102787/what-is-the-release-flag-in-the-java-9-compiler

fixes #1066

### CODE REVIEW

**Comment by johnou** [comment]
@jonohart this is actually failing on CI with the following..

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project atomix-utils: Fatal error compiling: invalid flag: --release -> [Help 1]
```

File: pom.xml [hunk 1]
```diff
@@ -318,6 +319,7 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>${maven.compiler.plugin.version}</version>
         <configuration>
+          <release>${java.releaseVersion}</release>
```
Comment (johnou) [inline pom.xml:322]: @jonohart can you add this as an override to the java9 profile?

**Comment by johnou** [comment]
Not sure what's up with jacoco and coveralls..

[ERROR] Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project atomix-parent: I/O operation failed: No coverage report files found -> [Help 1]

**Comment by johnou** [comment]
According to the JaCoCo documentation, do not set forkCount to 0 or set  forkMode to never as it would prevent executing the tests with the JaCoCo javaagent and no coverage would be recorded.

**Comment by johnou** [comment]
@jonohart thanks!

**Comment by jonohart** [comment]
No thank you, you did much more work on this than me! :)

So this now means that the image should work no matter what JDK version you use to build?

If so I think we'd need to release a new version of Atomix so that we have a docker image that works that we can use.

**Pull request merged**